rfcs/polyfill_testing.md

```diff
@@ -0,0 +1,150 @@
+# RFC 111: Polyfill Testing
+
+## Summary
+
+Add a mechanism for loading one or more polyfill implementations of a web
+platform feature and testing their correctness on existing web platform tests.
+
+## Details
+
+Polyfills are typically best effort implementations of browser features. It's
+not uncommon for them to be outdated, broken or just not to implement all parts
+of the specification they are polyfilling. Additionally, polyfills may only work
+with certain browsers or with certain libraries. This proposal allows testing
+polyfills using the same test suite we test browser compliance with. Being able
+to know which tests the polyfill passes or doesn't pass is useful information
+when developers are determining whether they can rely on it, just as they would
+use this information to know whether they could rely on the feature in the
+browser.
+
+In order to support testing a polyfill, a runtime argument is added which
+when used specifies a single local Javascript file to be injected into
+responses. The contents of the file will be added as a classic script (to ensure
+the polyfill runs before any tests) automatically to responses with a
+`text/html` or `application/xhtml+xml` MIME type after the `doctype` and `html`
```

Comment:
If I understand correctly, the PR uses html5lib also for `application/xhtml+xml` responses. This will work if the test is sufficiently HTML-syntax compatible, e.g. doesn't use namespace prefixes. But it can also break the test.

The script that is injected can't use `<` or `&`. This could be fixed by replacing those characters with `&lt;` and `&amp;` when the response type is `application/xhtml+xml`, or always injecting `/*<![CDATA[*/` and `/*]]>*/` (but then the script can't use `]]>`).

Reply:
Leaving this out for now sounds good to me!